### PR TITLE
metrics: add a config to unregister the unused metrics

### LINKED ISSUE #34989: Reduce the size of the metrics reported by TiDB
## Enhancement

As the number of metrics grows, the cost of extracting metrics data by Prometheus is also increasing. Since some of the metrics keys are not shown in grafana, we can provide a config option to eliminate them. By doing this, the size of data can be reduced and the costs could be saved.

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: close #34989

Problem Summary:

### What is changed and how it works?

- A boolean type config option `status.metrics-simplified` is added
- The size of `/metrics` can be decreased from 32K to 29K when the option is enabled.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)

The following test is based on [this commit](https://github.com/tangenta/tidb/commit/b36aa3bbd2c10f6b9fcd776063c1d7b43287fb54):
```console
// Default
[2022/05/26 20:47:07.449 +08:00] [INFO] [http_status.go:202] ["metric size"] [size=32062]

// With config option status.metrics-simplified = true
[2022/05/26 20:48:52.449 +08:00] [INFO] [http_status.go:202] ["metric size"] [size=29317]
```
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -822,6 +824,7 @@ var defaultConf = Config{
 		StatusHost:            DefStatusHost,
 		StatusPort:            DefStatusPort,
 		MetricsInterval:       15,
+		MetricsSimplified:     false,
```
Comment (bb7133) [inline config/config.go:827]: Please change this configuration to a global-scoped system variable, which is recommended now.

Comment (tangenta) [reply config/config.go]: Updated. PTAL

**Review by bb7133 — APPROVED** [review]
LGTM

**Review by CbcWestwolf — APPROVED** [review]

**Review by Defined2014 — APPROVED** [review]

**Pull request merged**